distributed/utils.py

```diff
@@ -386,7 +386,7 @@ def key_split_group(x):
     typ = type(x)
     if typ is tuple:
         return x[0]
-    elif typ is str:
+    elif typ is unicode or typ is str:
```

Comment:
On master:
```
>>> timeit.repeat("key_split_group('x-123')", setup="from distributed.utils import key_split_group")
[0.9205930233001709, 0.8670709133148193, 0.822983980178833]
>>> timeit.repeat("key_split_group(('x', 0, 1))", setup="from distributed.utils import key_split_group")
[0.4636390209197998, 0.4523329734802246, 0.4530479907989502]
```
On this branch:
```
>>> timeit.repeat("key_split_group('x-123')", setup="from distributed.utils import key_split_group")
[0.9638810157775879, 0.9432499408721924, 0.9448020458221436]
>>> timeit.repeat("key_split_group(('x', 0, 1))", setup="from distributed.utils import key_split_group")
[0.47734498977661133, 0.4366590976715088, 0.4450240135192871]
```
This is the time for running 1000000 executions, three times (the default settings for `timeit.repeat`).